src/Form/EmailFormProcessor.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Form;
+
+use App\Mail\ForgotPasswordMailer;
+use App\Repository\UserRepository;
+use Symfony\Component\Form\FormInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\Uid\Uuid;
+
+class EmailFormProcessor
+{
+    private UserRepository $userRepository;
+
+    private ForgotPasswordMailer $forgotPasswordMailer;
+
+    public function __construct(UserRepository $userRepository, ForgotPasswordMailer $forgotPasswordMailer)
+    {
+        $this->userRepository = $userRepository;
+        $this->forgotPasswordMailer = $forgotPasswordMailer;
+    }
+
+
+    public function processEmailForm(FormInterface $form, Request $request): void
```

Comment:
If you chose this approach you also need to make sure that the $form argument is valid. For example, I can send any form here that does not have the email field and your method will crash.